sound/soc/sof/sof-of-dev.c

```diff
@@ -77,6 +77,8 @@ int sof_of_probe(struct platform_device *pdev)
 		sof_pdata->fw_filename = desc->default_fw_filename[SOF_IPC];
 	}
 
+	sof_pdata->tplg_filename = desc->default_tplg_filename[SOF_IPC];
+
```

Comment:
that's a really interesting one. For Intel/AMD platforms, we determine the topology name when we identify the machine using all kinds of ACPI/DMI quirks.

I don't know if these is a parallel with OF/DT platforms since it's my understanding the topology name should probably be set when you probe the machine driver, no?

this feels like blurring layers in the OF/DT case. Clearly this doesn't scale if you have different codecs connected to the same SOC, the default tplg not would not be right except in one case :-)

Reply:
I think this is the same for i.MX. We identify the machine based on OF compatible string. And then based on this we use the proper `sof_dev_desc`.

```
static struct sof_dev_desc sof_of_imx8qxp_desc = {
	.ipc_supported_mask	= BIT(SOF_IPC),
	.ipc_default		= SOF_IPC,
	.default_fw_path = {
		[SOF_IPC] = "imx/sof",
	},
	.default_tplg_path = {
		[SOF_IPC] = "imx/sof-tplg",
	},
	.default_fw_filename = {
		[SOF_IPC] = "sof-imx8x.ri",
	},
	.default_tplg_filename = {
		[SOF_IPC] = "sof-imx8-wm8960.tplg",
	},
	.nocodec_tplg_filename = "sof-imx8-nocodec.tplg",
	.ops = &sof_imx8x_ops,
};
static struct sof_dev_desc sof_of_imx8qm_desc = {
	.ipc_supported_mask	= BIT(SOF_IPC),
	.ipc_default		= SOF_IPC,
	.default_fw_path = {
		[SOF_IPC] = "imx/sof",
	},
	.default_tplg_path = {
		[SOF_IPC] = "imx/sof-tplg",
	},
	.default_fw_filename = {
		[SOF_IPC] = "sof-imx8.ri",
	},
	.default_tplg_filename = {
		[SOF_IPC] = "sof-imx8-wm8960.tplg",
	},
	.nocodec_tplg_filename = "sof-imx8-nocodec.tplg",
	.ops = &sof_imx8_ops,
};

```